### DISCUSSION TITLE
Nested Provider/Repeater Issue

### DISCUSSION THREAD

**Discussion by dw-bayer** [Budibase/budibase#3525] [Help]
I've built a bit of a monstrosity of a page.

```
Provider A
  Repeater A
    Form
      Provider B (Data is a Table, not a relationship)
        Repeater B
          Provider C (Data is a relationship to Repeater B)
            Repeater C
              Provider D (data is a relationship to Repeater C)
                Repeater D
```
Everything is doing what I want until I get to Provider D. It is giving me a result of 0 rows, however, if I make an element with `Repeater C.ValuesThatIWantedToLoopThrough` it displays them as a single comma separated element. But I want to show them one at a time and with some other stuff :(

Let me know if I oversimplified this and left out crucial info...

Edit 1:
Is there a way to use handlebars to foreach that last bit of data? something like this (doesn't work but I'm really hoping it is because i'm doing it wrong).
```
{{ #forEach RepeaterC.dataC.ValuesThatIWantedToLoopThrough }}  {{ description }} {{ /forEach }}

**Comment by mjashanks** [comment]
Hey @dw-bayer. So is `ValuesThatIWantedToLoopThrough` a relationship? Or a multi-option?

This months release should be out by next week and will allow you to iterate through relationships, using Javascript (or handlebars - but javascript is easier).

Watch out for this one: #3151

**Reply by dw-bayer** [reply]
it's a relationship.

That release _does_ sound quite promising!
